src/data/stream.py

```diff
@@ -2,116 +2,109 @@
-        if cv2.waitKey(1) & 0xFF == ord("q"):
-            break
-
-
-def setup():
-    def set_video_cap_attrs(cap):
-        # only keep two frames in buffer. Supposedly improves speed.
-        cap.set(cv2.CAP_PROP_BUFFERSIZE, cap_buffer_size)
-
-        # set capture resolution (low-res).
-        cap.set(cv2.CAP_PROP_FRAME_WIDTH, cam_width)
-        cap.set(cv2.CAP_PROP_FRAME_HEIGHT, cam_height)
-
-    # opencv setup
-    left = cv2.VideoCapture(0)
-    right = cv2.VideoCapture(1)
-    set_video_cap_attrs(left)
-    set_video_cap_attrs(right)
-
-    # more mediapipe setup
-    hands = mphands.Hands()
-
-    # verify attributes as expected.
-    def test_video_cap_attrs(cap):
-        res = (
-            int(cap.get(cv2.CAP_PROP_BUFFERSIZE)) == cap_buffer_size
-            and int(cap.get(cv2.CAP_PROP_FRAME_WIDTH)) == cam_width
-            and int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT)) == cam_height
-        )
-        return res
-
-    assert (test_video_cap_attrs(left) and test_video_cap_attrs(right)) == True
-
-    return left, right, hands
-
-
-def teardown(left, right):
-    left.release()
-    right.release()
-
-    cv2.destroyAllWindows()
-
-
-def run():
-    left, right, hands = setup()
-
-    main(left, right, hands, old_frame_t, new_frame_t)
-
-    teardown(left, right)
-
+class InputStream:
+    def __init__(self):
+        # Initialize camera constants
+        self.CAP_BUFFER_SIZE = 2
+        self.CAMERA_WIDTH = 640
+        self.CAMERA_HEIGHT = 480
+        
+        #Initialize MediaPipe tools 
+        self.mp_drawing = mp.solutions.drawing_utils
+        self.mp_hands = mp.solutions.hands
+        self.hands = self.mp_hands.Hands()
+        
+        #Tracking for FPS calculation
+        self.old_frame_time = 0
+
+    def setVideoCaptureAttributes(self, cap):
+        # Returns True if camera attributes are successful
+        try:
+            # Sets buffer size and resolution for the camera capture.
+            # Reduces buffer size to 2 which reduces latency and improves  
+            # capture speed due to frames being processed more immediately.
+            cap.set(cv2.CAP_PROP_BUFFERSIZE, self.CAP_BUFFER_SIZE)
+            
+            # Sets capture resolution to low-res.
+            cap.set(cv2.CAP_PROP_FRAME_WIDTH, self.CAMERA_WIDTH)
+            cap.set(cv2.CAP_PROP_FRAME_HEIGHT, self.CAMERA_HEIGHT)
+            
+            return True
+        except:
+            return False
+
+    def setupCameras(self, numCameras):
+        # Initializes a specified number of cameras(usually 2) along with 
+        # there attributes and returns a list of VideoCapture objects.
+        cameras = []
+        for i in range(numCameras):
+            cap = cv2.VideoCapture(i)
+            if self.setVideoCaptureAttributes(cap):
+                cameras.append(cap)
+            else:
+                print(f"Failed to set up camera {i}")
+        return cameras
+
+    def process_frame(self, frame):
```

Comment:
Any functions not currently in the updated [interface design doc](https://github.com/HelloWorldSeniorProject/AirKeys/tree/develop/documentation/design/output) may be handled elsewhere & are not needed in this interface.